controller/controllercache/cache.go

```diff
@@ -0,0 +1,124 @@
+package controllercache
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	appv1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	cacheutil "github.com/argoproj/argo-cd/util/cache"
+)
+
+var ErrCacheMiss = cacheutil.ErrCacheMiss
+
+type Cache struct {
+	cache                           *cacheutil.Cache
+	appStateCacheExpiration         time.Duration
+	connectionStatusCacheExpiration time.Duration
+	oidcCacheExpiration             time.Duration
+}
+
+func NewCache(
+	cache *cacheutil.Cache,
+	appStateCacheExpiration time.Duration,
+	connectionStatusCacheExpiration time.Duration,
+	oidcCacheExpiration time.Duration,
+) *Cache {
+	return &Cache{cache, appStateCacheExpiration, connectionStatusCacheExpiration, oidcCacheExpiration}
+}
+
+type OIDCState struct {
+	// ReturnURL is the URL in which to redirect a user back to after completing an OAuth2 login
+	ReturnURL string `json:"returnURL"`
+}
+
+func AddCacheFlagsToCmd(cmd *cobra.Command) func() (*Cache, error) {
+	var appStateCacheExpiration time.Duration
+	var connectionStatusCacheExpiration time.Duration
+	var oidcCacheExpiration time.Duration
+
+	cmd.Flags().DurationVar(&appStateCacheExpiration, "app-state-cache-expiration", 1*time.Hour, "Cache expiration for app state")
```

Comment:
The cache package was split into two packages to avoid adding unnecessary flags to servers/controller binaries. But now controller still gets couple unused flags ( `connection-status-cache-expiration` and `oidc-cache-expiration` ) . I see a couple additional issues with this change: api-server now depends on `controller` code and `controller/controllercache` is not a great name.

I don't think we expect the user to modify any caching timeout except `--repo-cache-expiration`. Only this flag is documented. I suggest keeping `api-server`/`controller` related cache in `util/cache` and introduce flags only for repo server cache timeouts.

Reply:
maybe it was late in the day - I'll fix